src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs

```diff
@@ -125,6 +125,11 @@ public override void Initialize(AnalysisContext analysisContext)
 
             analysisContext.RegisterCompilationAction(compilationContext =>
             {
+                if (!(compilationContext.Compilation.SyntaxTrees.FirstOrDefault() is SyntaxTree tree))
```

Comment:
Simpler approach as code metrics options are unlikely to differ between documents. We can change this in future if required.